metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiHarvest.java

```diff
@@ -0,0 +1,64 @@
+package eu.europeana.metis.harvesting.oaipmh;
+
+import java.time.Instant;
+
+/**
+ * Immutable object representing a OAI-PMH harvest request.
+ */
+public class OaiHarvest extends OaiRepository {
+
+  private static final long serialVersionUID = -6300664626628834436L;
+
+  private final String setSpec;
+  private final Instant from;
+  private final Instant until;
+
+  /**
+   * Constructor.
+   *
+   * @param repositoryUrl The base url of the repository.
+   * @param metadataPrefix The metadata prefix (optional).
+   * @param setSpec The set spec value.
+   * @param from The from value (optional).
+   * @param until The until value (optional).
+   */
+  public OaiHarvest(String repositoryUrl, String metadataPrefix, String setSpec,
+          Instant from, Instant until) {
```

Comment:
Sugestion: Another good opportunity for Builder I believe

Reply:
Yes, this one I agree with.